### PR TITLE
Pass searchParams prop to head

### LINKED ISSUE #43269: head.js doesn't get searchParams
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```
    Operating System:
      Platform: darwin
      Arch: x64
      Version: Darwin Kernel Version 22.1.0: Sun Oct  9 20:15:09 PDT 2022; root:xnu-8792.41.9~2/RELEASE_ARM64_T6000
    Binaries:
      Node: 16.13.1
      npm: 8.1.2
      Yarn: 1.22.19
      pnpm: 7.9.5
    Relevant packages:
      next: 13.0.4
      eslint-config-next: 13.0.4
      react: 18.2.0
      react-dom: 18.2.0
```

### What browser are you using? (if relevant)

_No response_

### How are you deploying your application? (if relevant)

_No response_

### Describe the Bug

It seems `head.js` files doesn't get `searchParams` in props

### Expected Behavior

I would expect to get `searchParams` in props for `head.js` files the same way we have it in `page.js` files

### Link to reproduction - Issues with a link to complete (but minimal) reproduction code will be addressed faster

https://stackblitz.com/edit/vercel-next-js-zykntz?file=app/head.tsx

### To Reproduce

Check terminal panel

### PR DESCRIPTION
**Pull request by itrelease**
Fixes: [#43269](https://github.com/vercel/next.js/issues/43269)

## Feature

- [ ] Implements an existing feature request or RFC. Make sure the feature request has been accepted for implementation before opening a PR.
- [x] Related issues linked using `fixes #number`
- [x] [e2e](https://github.com/vercel/next.js/blob/canary/contributing/core/testing.md#writing-tests-for-nextjs) tests added
- [ ] Documentation added
- [ ] Telemetry added. In case of a feature if it's used or not.
- [ ] Errors have a helpful link attached, see [`contributing.md`](https://github.com/vercel/next.js/blob/canary/contributing.md)

## Documentation / Examples

- [x] Make sure the linting passes by running `pnpm build && pnpm lint`
- [ ] The "examples guidelines" are followed from [our contributing doc](https://github.com/vercel/next.js/blob/canary/contributing/examples/adding-examples.md)

### CODE REVIEW

File: test/e2e/check-searchparams-in-head/app/head.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+export default function HomePage({ searchParams }) {
```
[the remark this answers is not in the document]
Comment (timneutkens) [reply test/e2e/check-searchparams-in-head/app/head.js:1]: Specifically https://github.com/vercel/next.js/tree/canary/test/e2e/app-dir/head

**Review by timneutkens — CHANGES_REQUESTED** [review]
Blocking merge on a review from @sebmarkbage

File: test/e2e/check-searchparams-in-head/app/head.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (itrelease) [reply test/e2e/check-searchparams-in-head/app/head.js:1]: I've encountered weird behaviour. When I'm creating new nextjs app and link imports from source code I see that everything works, but when I run test `searchParams` are missing (not only for head files but also for page files) and `query` is always an empty object

**Comment by gaspardip** [comment]
> It should only be passed if the resolved head.ts file is at the same level as page.ts, but we have some new APIs coming that will solve this anyway.

mind to share a workaround while these new APIs are coming out?

**Pull request closed without merging**